backend/Functions/Edna.AssignmentLearnContent/Edna.AssignmentLearnContent/AssignmentLearnContentApi.cs

```diff
@@ -67,15 +81,36 @@ public AssignmentLearnContentApi(ILogger<AssignmentLearnContentApi> logger, IMap
         }
 
         [FunctionName(nameof(SaveAssignmentLearnContent))]
-        public void SaveAssignmentLearnContent(
+        public async Task<IActionResult> SaveAssignmentLearnContent(
             [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "assignments/{assignmentId}/learn-content/{contentUid}")] HttpRequest req,
-            [Table(AssignmentLearnContentTableName)] out AssignmentLearnContentEntity assignmentLearnContentEntity,
+            [Table(AssignmentLearnContentTableName)] IAsyncCollector<AssignmentLearnContentEntity> linksCollector,
             string assignmentId,
-            string contentUid)
+            string contentUid,
+            [Assignment(AssignmentId = "{assignmentId}")] Assignment assignment,
+            [Platform] PlatformsClient platformsClient,
+            [LtiAdvantage] INrpsClient nrpsClient,
+            [Lti1] Lti1MembershipClient membershipClient)
         {
+            IActionResult res = await ValidateUser(req, assignment, platformsClient, nrpsClient, membershipClient);
+            if (res.GetType() != typeof(OkResult))
+                return res;
+
             _logger.LogInformation($"Saving assignment learn content with uid [{contentUid}] to assignment {assignmentId}");
 
-            assignmentLearnContentEntity = new AssignmentLearnContentEntity { PartitionKey = assignmentId, RowKey = contentUid, ETag = "*" };
+            AssignmentLearnContentEntity assignmentLearnContentEntity = new AssignmentLearnContentEntity();
+            assignmentLearnContentEntity.PartitionKey = assignmentId;

```

Comment:
Is there no way to inline these into the constructor call itself?

Reply:
I had previously made some other changes here, because of which this had to be segregated. I must have missed to revert this. Thanks for pointing it out, I'll do the needful.